Blog Post: Advancements in Bitcoin and Lightning Wallets (#497)
OpenSats/website: data/blog/advancements-in-bitcoin-and-lightning-wallets.mdx

offline signing.

Through OpenSats' support, designer Adrian Lischer refined Cove's interface to
make advanced features more approachable. Navigation and transaction flows have
been simplified, and settings now enable switching between basic and advanced
modes.

By combining open-source libraries, hardware integration, and an intuitive user
experience, the project helps advance open Bitcoin development and keeps privacy
and usability in balance. Cove represents a practical evolution in mobile wallet
design, helping new users start safely, while giving experienced users greater
control, and reinforcing open standards that underpin the Bitcoin network.

[Cove]: https://covebitcoinwallet.com/
[support]: /blog/bitcoin-grants-july-2024#cove
[BDK]: https://bitcoindevkit.org/
[`TAPSIGNER`]: https://tapsigner.com/
[sign transactions]: https://github.com/notmandatory/rust-cktap/pull/29
[`BIP-329`]: https://github.com/bitcoin/bips/blob/master/bip-0329.mediawiki
[Swift]: https://github.com/bitcoinppl/bbqr-swift
[Android]: https://central.sonatype.com/artifact/org.bitcoinppl/bbqr-android/overview
[`BBQr` animated-QR protocol]: https://github.com/bitcoinppl/bbqr-ffi

> [Fully Noded]: https://github.com/Fonta1n3/FullyNoded

Author: ```suggestion
[Cove] is a single-signature mobile wallet for on-chain Bitcoin use. Unlike many
wallets that focus on Lightning or multisignature setups, Cove offers a simple,
self-custodial option compatible with hardware signing devices.

OpenSats' [support] helped developer Praveen Perera advance Cove from prototype
to production, releasing it as a non-custodial, Bitcoin-only wallet on the iOS
App Store. Written in `Rust` and integrated through Swift and Kotlin bindings
built on the Bitcoin Development Kit ([BDK]), the app supports [`TAPSIGNER`] and
[`SATSCARD`] hardware cards that let users initialize, change PINs, create
backups, and [sign transactions] directly through NFC.

Beyond Cove itself, Perera contributed upstream improvements to the Rust
ecosystem that make these features possible, including work on [`BIP-329`],
[`rust-cktap`], and [`uniffi`], which powers Rust bindings used by projects like
Cove and `BDK`. These upstream efforts directly strengthen Cove’s functionality
and the broader Bitcoin app ecosystem.

The wallet supports [`BIP-329`] labels, which let users name and organize
transactions for easier tracking. It includes detailed coin control, allowing
users to choose which coins to spend or save for better privacy and fee
management. Custom fee settings and local currency options provide clear
spending control. A decoy PIN mode unlocks a limited version of the wallet under
duress, adding security through plausible deniability.

> Cove isn't doing any one new thing—it's bringing all the right pieces
> together. Coin control on a phone is hard to do well, and I'm proud of how
> usable it turned out.
>
> <cite>—Praveen Perera</cite>

Cove also improved compatibility through [Swift] and [Android] bindings for the
[`BBQr` animated-QR protocol]. This system breaks large transaction files into
scannable QR sequences, allowing air-gapped hardware wallets to sign
transactions offline while keeping private keys isolated. Developers from other
projects, including [Fully Noded], have adopted this method to standardize
offline signing.

Through OpenSats' support, designer Adrian Lischer refined Cove's interface to
make advanced features more approachable. Navigation and transaction flows have
been simplified, and settings now enable switching between basic and advanced
modes.

By combining open-source libraries, hardware integration, and an intuitive user
experience, the project helps advance open Bitcoin development and keeps privacy
and usability in balance. Cove represents a practical evolution in mobile wallet
design, helping new users start safely, while giving experienced users greater
control, and reinforcing open standards that underpin the Bitcoin network.

[Cove]: https://covebitcoinwallet.com/
[BDK]: https://bitcoindevkit.org/
[`TAPSIGNER`]: https://tapsigner.com/
[`SATSCARD`]: https://coinkite.com/cards/satscard
[sign transactions]: https://github.com/notmandatory/rust-cktap/pull/29
[`BIP-329`]: https://github.com/bitcoinppl/bip329
[`rust-cktap`]: https://github.com/notmandatory/rust-cktap
[`uniffi`]: https://github.com/mozilla/uniffi-rs
[Swift]: https://github.com/bitcoinppl/bbqr-swift
[Android]: https://central.sonatype.com/artifact/org.bitcoinppl/bbqr-android/overview
[`BBQr` animated-QR protocol]: https://github.com/bitcoinppl/bbqr-ffi
[Fully Noded]: https://github.com/Fonta1n3/FullyNoded
```

---
title: "Advancements in Bitcoin and Lightning Wallets"
date: '2025-10-23'
tags: ['OpenSats', 'bitcoin', 'lightning', 'impact']
authors: ['default', 'arvin', 'dergigi']
images: ['/static/images/blog/86-advancements-in-bitcoin-and-lightning-wallets.jpg']
draft: false
summary: "An impact report from the front-lines of bitcoin and Lightning wallet development."
---

Wallets and node software form the core of the free and open financial ecosystem
that is Bitcoin. They define how people hold, send, and verify their bitcoin,
and how easily they can do so without depending on intermediaries. As the
network grows, these tools continue to evolve toward stronger privacy, smoother

> interoperability, and better usability across devices.

Author: Incorporated some edit suggestions provided by @IfA6Was9 🙏

Over the past year, a new wave of development has advanced both on-chain and
Lightning self-custody. Developers have improved the speed and reliability of
mobile and desktop wallets, added advanced features once reserved for experts,
and refined interfaces that make complex Bitcoin functions more intuitive. This
progress shows that usability and technical depth can advance together without
compromising security or control.

With support from OpenSats, the projects featured in this impact report
represent advances in Bitcoin and Lightning wallet development. Together, they
reinforce the open-source foundations that make self-custody practical and
resilient for users at every level of experience.

> The six projects highlighted in this impact report are:

Author: ```suggestion
Over the past year, a new wave of development has advanced both on-chain and
Lightning self-custody. Developers have improved the speed and reliability of
mobile and desktop wallets, added advanced features once reserved for experts,
and made complex Bitcoin functions more intuitive. This progress shows that ease
of use and technical depth can advance together without compromising security or
control.

With support from OpenSats, the six projects featured in this impact report
represent advances in Bitcoin and Lightning wallet development.

The projects highlighted in this impact report are:
```

Author: Incorporated some additional edit suggestions provided by @IfA6Was9 🙏

providing funding to several [ecash wallets](/blog/advancements-in-ecash).

These projects demonstrate the ongoing refinement of self-custody across the
Bitcoin ecosystem. From full-node operators to mobile-first users, developers
are closing the gap between technical sophistication and everyday reliability.
The result is a more capable and accessible network built on verifiable code and
open collaboration.

Let's take a closer look at how each of these projects is making an impact.

---

### Bitcoin Safe

[Bitcoin Safe] is a savings wallet built for secure, private, long-term
self-custody. It helps users manage their bitcoin holdings through a simple
interface while maintaining control of funds.

Funding [from OpenSats] enabled ongoing improvements in reliability and
transparency. The wallet now connects directly to Bitcoin's network for [live
transaction notifications], removing the need for centralized servers. Users can
see on-chain confirmations in real time while preserving privacy.

[Support] for hardware wallets like Blockstream Jade Plus expanded compatibility
and security. The wallet also [warns] about look-alike address scams and tracks

> [unconfirmed transactions] to reduce risk.

Reviewer: this one doesn't have a link?

Behind the interface, Bitcoin Safe is prepared for full adoption of [`Compact
Block Filters`] (`BIP157/158`), which let wallets identify their own

> transactions privately without querying external servers. This brings privacy

Reviewer: IDK how Bitcoin Safe works, but block filters do query from "external" servers. It just doesn't compromise your privacy to the third party server.

Author: Addressed. 🙏

closer to that of full nodes while remaining lightweight, requiring less storage
or processing power. Project developers collaborated with the Bitcoin Dev Kit
(`BDK`) community to refine handling of unconfirmed transactions, address
lookahead, and encrypted storage, helping improve BDK-based wallets overall.

> >Compact Block Filters offer fast performance while removing centralization

Reviewer: Fetching block filters is usually from a (centralized) server. Unless you are fetching the block filters from your own node. IDK if this is the case with this wallet.

>
><cite>—Andreas Griffin</cite>

Translation support helps make the wallet accessible to users in multiple
languages for broader usability, while the use of [nostr] enables private data
synchronization, label sharing, and multi-signature coordination across devices
through its "Sync & Chat" tool. This system reduces trust in centralized
infrastructure while allowing smooth collaboration between co-signers.

Bitcoin Safe brings modern Bitcoin libraries and decentralized communication
tools together in one application built for long-term savings, showing how
open-source development makes self-custody practical.

[Bitcoin Safe]: https://bitcoin-safe.org/en/
[from OpenSats]: /blog/tenth-wave-of-bitcoin-grants#bitcoin-safe
[live transaction notifications]: https://github.com/andreasgriffin/bitcoin-p2p
[Support]: https://github.com/andreasgriffin/bitcoin-safe/releases/tag/1.1.0
[warns]: https://github.com/andreasgriffin/bitcoin-safe/releases/tag/1.2.2
[`Compact Block Filters`]: https://bitcoinops.org/en/topics/compact-block-filters/
[nostr]: https://github.com/nostr-protocol/nostr

### Blixt Wallet

[Blixt Wallet] is a non-custodial Bitcoin Lightning wallet for Android and iOS,

> focused on usability and control. Built with `LND` and `Neutrino`, it lets users

Reviewer: ```suggestion
focused on usability and control. Build with `LND` and `Neutrino`, it lets users
```
present form better I think?

trajectory.

[Blixt Wallet]: https://blixtwallet.github.io/
[a grant from OpenSats]: /blog/bitcoin-grants-july-2023#blixt-wallet
[`react-native-turbo-lnd`]: https://github.com/hsjoberg/react-native-turbo-lnd
[`react-native-turbo-sqlite`]: https://github.com/hsjoberg/react-native-turbo-sqlite
[`react-native-nitro-tor`]: https://github.com/niteshbalusu11/react-native-nitro-tor
[updates to Neutrino]: https://github.com/lightninglabs/neutrino/pull/300
[version 0.6.9-420]: https://github.com/hsjoberg/blixt-wallet/releases/tag/v0.6.9-420
[`bLIP`]: https://github.com/hsjoberg/blips/blob/lnurl-forwarding/blip-lnurlforward.md

### Blitz Wallet

[Blitz Wallet] is a self-custodial Bitcoin wallet that unifies Lightning,
Liquid, and ecash balances in one app. Its goal is to make Bitcoin payments
accessible while maintaining full control of funds.

Built with the `Breez SDK`, `Blockstream Greenlight`, and the `Blockstream Green
Development Kit`, Blitz combines these open-source tools to provide a strong
self-custodial foundation while using Boltz for automatic channel rebalancing.
With OpenSats' [support], Blitz evolved from a hybrid prototype into a

> multi-protocol wallet. The latest version migrates to the Spark protocol, which

Reviewer 2: Keeping with the similar theme of the last comment.

Maybe change:
```
Built with the Breez SDK, Blockstream Greenlight, and the Blockstream Green Development Kit, Blitz combines these open-source tools to provide a strong self-custodial foundation while using Boltz for automatic channel rebalancing. With OpenSats' support, Blitz evolved from a hybrid prototype into a multi-protocol wallet. The latest version migrates to the Spark protocol, which enables receiving payments while offline, maintaining inbound capacity, and exiting directly to on-chain bitcoin. A new Progressive Web App version works on any browser without relying on the app store.
```

To something like:
```
Over the last year, Blitz pioneered a new "custodial ladder approach" to onboarding new Bitcoiners by using tools such as the `Breez SDK`, `Blockstream Greenlight`, and the `Blockstream Green Development Kit`. Blitz combined these open-source tools to provide a strong self-custodial foundation while using Boltz for automatic channel rebalancing. However, with balance fragmentation and educational challenges, the framework was ultimately too complex to call “simple.” With OpenSats' [support], Blitz evolved from this hybrid, complex prototype into a truly simple wallet. The latest version migrates to the Spark protocol, which enables receiving payments while offline, maintaining inbound capacity, and
exiting directly to on-chain bitcoin. A new [`Progressive Web App`] version
[works on any browser] without relying on the app store.
```

control that define Bitcoin self-custody.

[Satsigner]: https://satsigner.com/
[funding from OpenSats]: /blog/bitcoin-grants-feb-2024#satsigner
[descriptor-native multisignature]: https://github.com/satsigner/satsigner/commit/bb47b855b44bd9bffb7d18b57c64616a1b202b64
[`Electrum`]: https://github.com/spesmilo/electrum
[`Sankey`]: https://en.wikipedia.org/wiki/Sankey_diagram
[Sparrow]: https://sparrowwallet.com/
[Specter]: https://specter.solutions/

---

Each of these projects advances the usability of bitcoin without compromising
self-custody. By improving Lightning reliability, refining wallet interfaces,
and strengthening privacy, they make it easier for users to hold and transact
securely on their own terms.

OpenSats supports this work to ensure that critical infrastructure remains open,
verifiable, and decentralized. The developers behind these tools strengthen
Bitcoin's foundation by turning technical progress into practical freedom,
empowering anyone, anywhere, to use bitcoin with confidence and autonomy.

Together, these efforts demonstrate the steady evolution of Bitcoin as both
technology and public good. Self-custody is becoming simpler, more intuitive,

> and more resilient. Each release brings the network closer to a future where

Reviewer: ```suggestion
and more resilient. Each release brings Bitcoin closer to a future where
```
nit, wallet software like these ones do not impact "the network"
